src/node_http_parser.cc

```diff
@@ -289,13 +287,7 @@ class Parser : public AsyncWrap, public StreamListener {
     Local<Value> cb = object()->Get(env()->context(), kOnMessageBegin)
                               .ToLocalChecked();
     if (cb->IsFunction()) {
-      InternalCallbackScope callback_scope(
-        this, InternalCallbackScope::kSkipTaskQueues);
-
-      MaybeLocal<Value> r = cb.As<Function>()->Call(
-        env()->context(), object(), 0, nullptr);
-
-      if (r.IsEmpty()) callback_scope.MarkAsFailed();
+      USE(cb.As<Function>()->Call(env()->context(), object(), 0, nullptr));
```

Comment:
We should avoid using `USE(...)` here as it does not allow thrown errors to propagate correctly here. Note the previous code had the check for `IsEmpty()` and setting `MarkAsFailed`... that behavior should be retained.

Reply:
If there aren't any such tests already, a test verifying that errors thrown in HTTPParser callbacks are handled correctly would be helpful in verifying the changed behavior here.